internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -339,13 +353,103 @@ func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
 
 // IsContainerRunning checks if a specific Docker container is running.
 func (c DockerCmdClient) IsContainerRunning(ctx context.Context, name string) (bool, error) {
+	state, err := c.containerState(ctx, name)
+	if err != nil {
+		return false, err
+	}
+	switch state.Status {
+	case containerStatusRunning:
+		return true, nil
+	case containerStatusExited:
+		return false, &ErrContainerExited{name: name, exitcode: state.ExitCode}
+	}
+	return false, nil
+}
+
+// IsContainerCompleteOrSuccess returns true if a docker container exits with an exitcode.
+func (c DockerCmdClient) IsContainerCompleteOrSuccess(ctx context.Context, containerName string) (bool, int, error) {
+	state, err := c.containerState(ctx, containerName)
+	if err != nil {
+		return false, 0, err
+	}
+	return state.Status == containerStatusExited, state.ExitCode, nil
+}
+
+// IsContainerHealthy returns true if a container health state is healthy.
+func (c DockerCmdClient) IsContainerHealthy(ctx context.Context, containerName string) (bool, error) {
+	state, err := c.containerState(ctx, containerName)
+	if err != nil {
+		return false, err
+	}
+	if state.Status == containerStatusExited {
+		return false, &ErrContainerExited{name: containerName, exitcode: state.ExitCode}
+	}
+	if state.Health == nil {
+		return false, fmt.Errorf("healthcheck is not configured for container %s", containerName)
+	}
+	switch state.Health.Status {
+	case healthy:
+		return true, nil
+	case starting:
+		return false, nil
+	case unhealthy:
+		return false, fmt.Errorf("container %q is %q", containerName, unhealthy)
+	case noHealthcheck:
+		return false, fmt.Errorf("healthcheck configuration is set to %q for container %s", noHealthcheck, containerName)
```

Comment:
We should probably return the same error as https://github.com/aws/copilot-cli/pull/5488/files#diff-245a109ace8ab38c02069b4557e8256a2fcf479ab23323d871df8357ecce6b51R388